File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/Portfolios.cs
using System;
using System.IO;
using Xfinium.Pdf;
using Xfinium.Pdf.Graphics;
using Xfinium.Pdf.Portfolios;
using Xfinium.Pdf.Core.Cos;

namespace Xfinium.Pdf.Samples
{
    /// <summary>
    /// Portfolios sample.
    /// </summary>
    public class Portfolios
    {
        /// <summary>
        /// Main method for running the sample.
        /// </summary>
        /// <param name="imageStream"></param>
        /// <param name="pdfStream"></param>
        /// <param name="csStream"></param>
        /// <param name="vbStream"></param>
        public static SampleOutputInfo[] Run(Stream imageStream, Stream pdfStream, Stream csStream, Stream vbStream)
        {
            PdfPortfolio portfolio = new PdfPortfolio();

            // Build the structure that describes how to files and folders in the portfolio are presented to the user.
            PdfPortfolioAttributeDefinitions portfolioAttributeDefinitions = new PdfPortfolioAttributeDefinitions();
            PdfPortfolioAttributeDefinition nameAttribute = new PdfPortfolioAttributeDefinition();
            nameAttribute.Name = "Name";
            nameAttribute.Type = PdfPortfolioAttributeDefinitionType.String;
            portfolioAttributeDefinitions["name"] = nameAttribute;
            PdfPortfolioAttributeDefinition typeAttribute = new PdfPortfolioAttributeDefinition();
            typeAttribute.Name = "Type";
            typeAttribute.Type = PdfPortfolioAttributeDefinitionType.String;
            portfolioAttributeDefinitions["type"] = typeAttribute;

            portfolio.AttributeDefinitions = portfolioAttributeDefinitions;

            // Setup the folders structure
            PdfPortfolioFolder root = new PdfPortfolioFolder();
            root.Name = "All files";
            root.PortfolioAttributes = new PdfPortfolioItemAttributes();
            root.PortfolioAttributes["name"] = new PdfCosString("All files");

            PdfPortfolioFolder imagesFolder = new PdfPortfolioFolder();
            imagesFolder.Name = "Images";
            imagesFolder.PortfolioAttributes = new PdfPortfolioItemAttributes();
            imagesFolder.PortfolioAttributes["name"] = new PdfCosString("Images (1)");
            root.Folders.Add(imagesFolder);

            PdfPortfolioFolder pdfFolder = new PdfPortfolioFolder();
            pdfFolder.Name = "PDFs";
            pdfFolder.PortfolioAttributes = new PdfPortfolioItemAttributes();
            pdfFolder.PortfolioAttributes["name"] = new PdfCosString("PDFs (1)");
            root.Folders.Add(pdfFolder);

            PdfPortfolioFolder htmlFolder = new PdfPortfolioFolder();
            htmlFolder.Name = "HTML";
            htmlFolder.PortfolioAttributes = new PdfPortfolioItemAttributes();
            htmlFolder.PortfolioAttributes["name"] = new PdfCosString("HTML (2)");
            root.Folders.Add(htmlFolder);

            portfolio.Folders.Add(root);

            // Setup the portfolio items
            PdfPortfolioItem imageFile = new PdfPortfolioItem();
            imageFile.Folder = imagesFolder;
            byte[] data = new byte[imageStream.Length];
            imageStream.Read(data, 0, data.Length);
            imageFile.Payload = data;
            imageFile.FileName = "image.jpg";
            imageFile.Attributes = new PdfPortfolioItemAttributes();
            imageFile.Attributes["name"] = new PdfCosString("image.jpg");
            imageFile.Attributes["type"] = new PdfCosString("JPEG image");
            portfolio.Items.Add(imageFile);

            PdfPortfolioItem pdfFile = new PdfPortfolioItem();
            pdfFile.Folder = pdfFolder;
            data = new byte[pdfStream.Length];
            pdfStream.Read(data, 0, data.Length);
            pdfFile.Payload = data;
            pdfFile.FileName = "content.pdf";
            pdfFile.Attributes = new PdfPortfolioItemAttributes();
            pdfFile.Attributes["name"] = new PdfCosString("content.pdf");
            pdfFile.Attributes["type"] = new PdfCosString("PDF file");
            portfolio.Items.Add(pdfFile);

            PdfPortfolioItem csFile = new PdfPortfolioItem();
            csFile.Folder = htmlFolder;
            data = new byte[csStream.Length];
            csStream.Read(data, 0, data.Length);
            csFile.Payload = data;
            csFile.FileName = "portfolios.cs.html";
            csFile.Attributes = new PdfPortfolioItemAttributes();
            csFile.Attributes["name"] = new PdfCosString("portfolios.cs.html");
            csFile.Attributes["type"] = new PdfCosString("HTML file");
            portfolio.Items.Add(csFile);

            PdfPortfolioItem vbFile = new PdfPortfolioItem();
            vbFile.Folder = htmlFolder;
            data = new byte[vbStream.Length];
            vbStream.Read(data, 0, data.Length);
            vbFile.Payload = data;
            vbFile.FileName = "portfolios.vb.html";
            vbFile.Attributes = new PdfPortfolioItemAttributes();
            vbFile.Attributes["name"] = new PdfCosString("portfolios.vb.html");
            vbFile.Attributes["type"] = new PdfCosString("HTML file");
            portfolio.Items.Add(vbFile);

            portfolio.StartupDocument = pdfFile;

            SampleOutputInfo[] output = new SampleOutputInfo[] { new SampleOutputInfo(portfolio, "xfinium.pdf.sample.portfolios.pdf") };
            return output;
        }
    }
}

File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/Portfolios.vb
Imports System.IO
Imports Xfinium.Pdf
Imports Xfinium.Pdf.Graphics
Imports Xfinium.Pdf.Portfolios
Imports Xfinium.Pdf.Core.Cos

Namespace Xfinium.Pdf.Samples
	''' <summary>
	''' Portfolios sample.
	''' </summary>
	Public Class Portfolios
		''' <summary>
		''' Main method for running the sample.
		''' </summary>
		''' <param name="imageStream"></param>
		''' <param name="pdfStream"></param>
		''' <param name="csStream"></param>
		''' <param name="vbStream"></param>
		Public Shared Function Run(imageStream As Stream, pdfStream As Stream, csStream As Stream, vbStream As Stream) As SampleOutputInfo()
			Dim portfolio As New PdfPortfolio()

			' Build the structure that describes how to files and folders in the portfolio are presented to the user.
			Dim portfolioAttributeDefinitions As New PdfPortfolioAttributeDefinitions()
			Dim nameAttribute As New PdfPortfolioAttributeDefinition()
			nameAttribute.Name = "Name"
			nameAttribute.Type = PdfPortfolioAttributeDefinitionType.[String]
			portfolioAttributeDefinitions("name") = nameAttribute
			Dim typeAttribute As New PdfPortfolioAttributeDefinition()
			typeAttribute.Name = "Type"
			typeAttribute.Type = PdfPortfolioAttributeDefinitionType.[String]
			portfolioAttributeDefinitions("type") = typeAttribute

			portfolio.AttributeDefinitions = portfolioAttributeDefinitions

			' Setup the folders structure
			Dim root As New PdfPortfolioFolder()
			root.Name = "All files"
			root.PortfolioAttributes = New PdfPortfolioItemAttributes()
			root.PortfolioAttributes("name") = New PdfCosString("All files")

			Dim imagesFolder As New PdfPortfolioFolder()
			imagesFolder.Name = "Images"
			imagesFolder.PortfolioAttributes = New PdfPortfolioItemAttributes()
			imagesFolder.PortfolioAttributes("name") = New PdfCosString("Images (1)")
			root.Folders.Add(imagesFolder)

			Dim pdfFolder As New PdfPortfolioFolder()
			pdfFolder.Name = "PDFs"
			pdfFolder.PortfolioAttributes = New PdfPortfolioItemAttributes()
			pdfFolder.PortfolioAttributes("name") = New PdfCosString("PDFs (1)")
			root.Folders.Add(pdfFolder)

			Dim htmlFolder As New PdfPortfolioFolder()
			htmlFolder.Name = "HTML"
			htmlFolder.PortfolioAttributes = New PdfPortfolioItemAttributes()
			htmlFolder.PortfolioAttributes("name") = New PdfCosString("HTML (2)")
			root.Folders.Add(htmlFolder)

			portfolio.Folders.Add(root)

			' Setup the portfolio items
			Dim imageFile As New PdfPortfolioItem()
			imageFile.Folder = imagesFolder
			Dim data As Byte() = New Byte(imageStream.Length - 1) {}
			imageStream.Read(data, 0, data.Length)
			imageFile.Payload = data
			imageFile.FileName = "image.jpg"
			imageFile.Attributes = New PdfPortfolioItemAttributes()
			imageFile.Attributes("name") = New PdfCosString("image.jpg")
			imageFile.Attributes("type") = New PdfCosString("JPEG image")
			portfolio.Items.Add(imageFile)

			Dim pdfFile As New PdfPortfolioItem()
			pdfFile.Folder = pdfFolder
			data = New Byte(pdfStream.Length - 1) {}
			pdfStream.Read(data, 0, data.Length)
			pdfFile.Payload = data
			pdfFile.FileName = "content.pdf"
			pdfFile.Attributes = New PdfPortfolioItemAttributes()
			pdfFile.Attributes("name") = New PdfCosString("content.pdf")
			pdfFile.Attributes("type") = New PdfCosString("PDF file")
			portfolio.Items.Add(pdfFile)

			Dim csFile As New PdfPortfolioItem()
			csFile.Folder = htmlFolder
			data = New Byte(csStream.Length - 1) {}
			csStream.Read(data, 0, data.Length)
			csFile.Payload = data
			csFile.FileName = "portfolios.cs.html"
			csFile.Attributes = New PdfPortfolioItemAttributes()
			csFile.Attributes("name") = New PdfCosString("portfolios.cs.html")
			csFile.Attributes("type") = New PdfCosString("HTML file")
			portfolio.Items.Add(csFile)

			Dim vbFile As New PdfPortfolioItem()
			vbFile.Folder = htmlFolder
			data = New Byte(vbStream.Length - 1) {}
			vbStream.Read(data, 0, data.Length)
			vbFile.Payload = data
			vbFile.FileName = "portfolios.vb.html"
			vbFile.Attributes = New PdfPortfolioItemAttributes()
			vbFile.Attributes("name") = New PdfCosString("portfolios.vb.html")
			vbFile.Attributes("type") = New PdfCosString("HTML file")
			portfolio.Items.Add(vbFile)

			portfolio.StartupDocument = pdfFile

			Dim output As SampleOutputInfo() = New SampleOutputInfo() {New SampleOutputInfo(portfolio, "xfinium.pdf.sample.portfolios.pdf")}
			Return output
		End Function
	End Class
End Namespace
